### ISSUE TITLE
Binance: logging into Binance with wrong credentials throws a ValidationError exception

### ISSUE THREAD

**Issue by davidsblom** [nautechsystems/nautilus_trader#1161]
# Bug Report

### Expected Behavior
Error message saying that the login credentials are incorrect.

### Actual Behavior
```
2023-06-30T19:45:39.277381000Z [DBG] FIREHOSE-SPOT-001.DataEngine: Data queue processing starting (qsize=0)...
2023-06-30T19:45:39.277383000Z [DBG] FIREHOSE-SPOT-001.RiskEngine: Command message queue processing (qsize=0)...
2023-06-30T19:45:39.277384000Z [DBG] FIREHOSE-SPOT-001.RiskEngine: Event message queue processing starting (qsize=0)...
2023-06-30T19:45:39.277386000Z [DBG] FIREHOSE-SPOT-001.ExecEngine: Command message queue processing starting (qsize=0)...
2023-06-30T19:45:39.277387000Z [DBG] FIREHOSE-SPOT-001.ExecEngine: Event message queue processing starting (qsize=0)...
2023-06-30T19:45:39.277397000Z [INF] FIREHOSE-SPOT-001.DataClient-BINANCE: Initialising instruments...
2023-06-30T19:45:39.277399000Z [INF] FIREHOSE-SPOT-001.BinanceSpotInstrumentProvider: Loading all instruments...
2023-06-30T19:45:39.277400000Z [WRN] FIREHOSE-SPOT-001.BinanceSpotInstrumentProvider: Currently not requesting actual trade fees. All instruments will have zero fees.
2023-06-30T19:45:41.276658000Z [DBG] FIREHOSE-SPOT-001.ExecEngine: Checking in-flight orders status...
2023-06-30T19:45:41.276677000Z [DBG] FIREHOSE-SPOT-001.ExecEngine: Found 0 orders in-flight.
2023-06-30T19:45:41.937620000Z [ERR] FIREHOSE-SPOT-001.DataClient-BINANCE: Error on `_connect`: ValidationError("Invalid enum value 'TRD_GRP_010' - at `$.symbols[0].permissions[6]`")
2023-06-30T19:45:43.276660000Z [DBG] FIREHOSE-SPOT-001.ExecEngine: Checking in-flight orders status...
```

### Steps to Reproduce the Problem

1. Start a strategy listening to Binance data
2.
3.

### Specifications

- OS platform: Mac OS 
- Python version: 3.10
- `nautilus_trader` version: git commit 9993787

**Comment by cjdsellers** [comment]
I think the issue is that we didn't have an enum member value matching what was parsed `'TRD_GRP_010'`. The `msgspec` library is very strict on this.

This is at least the third time this has happened on this enum, the last time I extended the enum up to `'TRD_GRP_009'` thinking this would give us some breathing room, but seems they're now exceeding this number already. Its likely published in their release notes, but hard to keep up with.

I've now extended the values to `'TRD_GRP_020'` and pushed to `develop` branch.
